Header header
float32[]  x_solutions